code/src/utils/validators/productScheme.ts

```diff
@@ -0,0 +1,87 @@
+import { z } from "zod";
+
+import { ProductTranslation } from "@/containers/forms/new-product-form/NewProductForm.types";
+
+const validateProductLanguageFields = (data: ProductTranslation) => {
+  const isDescriptionEmpty = !data.description?.trim();
+  const isNameEmpty = !data.name?.trim();
+
+  return isNameEmpty === isDescriptionEmpty;
+};
+
+const getProductLanguageFieldsError = (
+  data: ProductTranslation,
+  index: number
+) => {
+  const isDescriptionEmpty = !data.description?.trim();
+  const isNameEmpty = !data.name?.trim();
+
+  if (!isNameEmpty && isDescriptionEmpty) {
+    return {
+      message:
+        "Either both name and description must be filled, either none of them",
+      path: [index, "description"]
+    };
+  }
+
+  return {
+    message:
+      "Either both name and description must be filled, either none of them",
+    path: [index, "name"]
+  };
+};
+
+const productTranslationSchema = z.object({
+  name: z
+    .string({ message: "Name cannot be empty" })
+    .min(3, { message: "Name is too short" })
+    .max(200, { message: "Name is too long" })
+    .optional()
+    .or(z.literal("")),
+  description: z
+    .string({ message: "Description cannot be empty" })
+    .min(10, { message: "Description is too short" })
+    .max(1000, { message: "Description is too long" })
+    .optional()
+    .or(z.literal("")),
+  languageCode: z.string({ message: "Language code cannot be empty" })
+});
+
+const productScheme = z.object({
+  image: z.string().url({ message: "Please provide a valid URL" }),
+  price: z
+    .number({ message: "Invalid price" })
+    .min(0, { message: "Must be bigger than 0" })
+    .max(100000000, { message: "Must be less than 100 mln" }),
+  status: z.boolean(),
+  quantity: z
+    .number({ message: "Invalid quantity" })
+    .min(0, { message: "Must be bigger than 0" })
+    .max(100000000, { message: "Must be less than 100 mln" }),
+  category: z.number({ message: "Please, select a category" }),
```

Comment:
Great work! @GeorgeShvab 
Should be translations here for `massages` ? What do you think ?

Reply:
As far as I sow, there is no translations for any of the form errors yet (signing and signup forms). I think, if we will consider to add it, we need to make it in separate pr so we will ad all of them